example/timer/odp_timer_accuracy.c

```diff
@@ -1107,63 +1177,92 @@ int main(int argc, char *argv[])
 	/* Configure scheduler */
 	odp_schedule_config(NULL);
 
-	test_global.timer_ctx = calloc(test_global.alloc_timers,
-				       sizeof(timer_ctx_t));
+	odp_shm_t shm = ODP_SHM_INVALID;
+	void *addr;
+	uint64_t size = sizeof(test_shm_t);
 
-	if (test_global.timer_ctx == NULL) {
-		printf("Timer context table calloc failed.\n");
-		ret = -1;
-		goto quit;
-	}
+	size += test_global.alloc_timers * sizeof(timer_ctx_t);
 
 	if (test_global.opt.output) {
-		odp_shm_t shm;
-		void *addr;
-		uint64_t size = test_global.tot_timers * sizeof(test_log_t);
-
 		test_global.file = fopen(test_global.filename, "w");
 		if (test_global.file == NULL) {
 			printf("Failed to open output file %s: %s\n",
 			       test_global.filename, strerror(errno));
 			ret = -1;
 			goto quit;
 		}
+		size += (test_global.tot_timers - test_global.warmup_timers) * sizeof(test_log_t);
+	}
 
-		shm = odp_shm_reserve("timer_accuracy_log", size,
-				      sizeof(test_log_t), 0);
+	shm = odp_shm_reserve("timer_accuracy", size,
+			      ODP_CACHE_LINE_SIZE, ODP_SHM_SINGLE_VA);
 
-		if (shm == ODP_SHM_INVALID) {
-			printf("Test log alloc failed.\n");
-			ret = -1;
-			goto quit;
-		}
-
-		addr = odp_shm_addr(shm);
-		memset(addr, 0, size);
-		test_global.log = addr;
-		test_global.log_shm = shm;
+	if (shm == ODP_SHM_INVALID) {
+		printf("Test log alloc failed.\n");
+		ret = -1;
+		goto quit;
 	}
 
+	addr = odp_shm_addr(shm);
+	memset(addr, 0, size);
+	test_global.shm = addr;
+	test_global.shm->timer_ctx = (timer_ctx_t *)((test_shm_t *)addr + 1);
+	if (test_global.opt.output)
+		test_global.shm->log =
+			(test_log_t *)&test_global.shm->timer_ctx[test_global.alloc_timers];
```

Comment:
The log can remain as a separate SHM. It simplifies code when two SHMs are used (avoids dependency between timer_ctx and log). It also decreases the lines modified by this patch (more focus on the changes actually needed).